src/test/java/it/gov/pagopa/payment/service/QRCodePaymentServiceImplTest.java

```diff
@@ -0,0 +1,84 @@
+package it.gov.pagopa.payment.service;
+
+import it.gov.pagopa.payment.dto.mapper.TransactionInProgress2SyncTrxStatus;
+import it.gov.pagopa.payment.dto.qrcode.SyncTrxStatusDTO;
+import it.gov.pagopa.payment.enums.SyncTrxStatus;
+import it.gov.pagopa.payment.exception.ClientExceptionNoBody;
+import it.gov.pagopa.payment.model.TransactionInProgress;
+import it.gov.pagopa.payment.repository.TransactionInProgressRepository;
+import it.gov.pagopa.payment.service.qrcode.QRCodeAuthPaymentService;
+import it.gov.pagopa.payment.service.qrcode.QRCodeConfirmationService;
+import it.gov.pagopa.payment.service.qrcode.QRCodeCreationService;
+import it.gov.pagopa.payment.service.qrcode.QRCodePreAuthService;
+import it.gov.pagopa.payment.test.fakers.SyncTrxStatusFaker;
+import it.gov.pagopa.payment.test.fakers.TransactionInProgressFaker;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import java.util.Optional;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class QRCodePaymentServiceImplTest {
+    @Mock
+    TransactionInProgressRepository transactionInProgressRepository;
+    @Mock
+    QRCodeCreationService qrCodeCreationService;
+    @Mock
+     QRCodePreAuthService qrCodePreAuthService;
+    @Mock
+    QRCodeAuthPaymentService qrCodeAuthPaymentService;
+    @Mock
+    QRCodeConfirmationService qrCodeConfirmationService;
+    @Mock
+    TransactionInProgress2SyncTrxStatus transactionMapper;
```

Comment:
A mapper is a so simple class (having a separate test which will guarantee for its right behavior) whose nature give us more benefits using its real implementation than its mock

We could trust its implementation (because there is a test that proof it) so could avoid to configure its mock, so let's just use the real implementation